### ISSUE TITLE
java.lang.ClassNotFoundException: Didn't find class "io.realm.DefaultRealmModuleMediator"

### ISSUE THREAD

**Issue by jgonfer** [realm/realm-java#1189]
I'm getting this error in my current project, but I started a new one based on a Blank Activity Template and when I try to run the app I get all this info:

06-06 12:42:17.335  23811-23811/com.jgonfer.b2b E/AndroidRuntime﹕ FATAL EXCEPTION: main
    Process: com.jgonfer.b2b, PID: 23811
    java.lang.ExceptionInInitializerError
            at com.jgonfer.b2b.StationsActivity.onCreate(StationsActivity.java:18)
            at android.app.Activity.performCreate(Activity.java:6289)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2646)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2760)
            at android.app.ActivityThread.access$900(ActivityThread.java:177)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1448)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:145)
            at android.app.ActivityThread.main(ActivityThread.java:5944)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1389)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1184)
     Caused by: io.realm.exceptions.RealmException: Could not find io.realm.DefaultRealmModuleMediator
            at io.realm.Realm.getDefaultMediator(Realm.java:256)
            at io.realm.Realm.<clinit>(Realm.java:165)
            at com.jgonfer.b2b.StationsActivity.onCreate(StationsActivity.java:18)
            at android.app.Activity.performCreate(Activity.java:6289)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2646)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2760)
            at android.app.ActivityThread.access$900(ActivityThread.java:177)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1448)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:145)
            at android.app.ActivityThread.main(ActivityThread.java:5944)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1389)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1184)
     Caused by: java.lang.ClassNotFoundException: io.realm.DefaultRealmModuleMediator
            at java.lang.Class.classForName(Native Method)
            at java.lang.Class.forName(Class.java:308)
            at java.lang.Class.forName(Class.java:272)
            at io.realm.Realm.getDefaultMediator(Realm.java:251)
            at io.realm.Realm.<clinit>(Realm.java:165)
            at com.jgonfer.b2b.StationsActivity.onCreate(StationsActivity.java:18)
            at android.app.Activity.performCreate(Activity.java:6289)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2646)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2760)
            at android.app.ActivityThread.access$900(ActivityThread.java:177)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1448)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:145)
            at android.app.ActivityThread.main(ActivityThread.java:5944)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1389)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1184)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "io.realm.DefaultRealmModuleMediator" on path: DexPathList[[zip file "/data/app/com.jgonfer.b2b-1/base.apk"],nativeLibraryDirectories=[/data/app/com.jgonfer.b2b-1/lib/arm, /vendor/lib, /system/lib]]
            at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
            at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
            at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
            at java.lang.Class.classForName(Native Method)
            at java.lang.Class.forName(Class.java:308)
            at java.lang.Class.forName(Class.java:272)
            at io.realm.Realm.getDefaultMediator(Realm.java:251)
            at io.realm.Realm.<clinit>(Realm.java:165)
            at com.jgonfer.b2b.StationsActivity.onCreate(StationsActivity.java:18)
            at android.app.Activity.performCreate(Activity.java:6289)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2646)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2760)
            at android.app.ActivityThread.access$900(ActivityThread.java:177)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1448)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:145)
            at android.app.ActivityThread.main(ActivityThread.java:5944)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1389)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1184)
    Suppressed: java.lang.ClassNotFoundException: io.realm.DefaultRealmModuleMediator
            at java.lang.Class.classForName(Native Method)
            at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
            at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
            at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
            ... 20 more
     Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack available

In my code, I simply call an instance of Realm like that:
```java
@Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_stations);

        Realm realm = Realm.getInstance(this);
    }
```
My build.gradle file:
```java
apply plugin: 'com.android.application'

android {
    compileSdkVersion 22
    buildToolsVersion "21.1.2"

    defaultConfig {
        applicationId "com.jgonfer.b2b"
        minSdkVersion 15
        targetSdkVersion 22
        versionCode 1000
        versionName "1.0"
    }
    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }
}
dependencies {
    compile fileTree(dir: 'libs', include: ['*.jar'])
    compile 'com.android.support:appcompat-v7:22.2.0'
    compile 'io.realm:realm-android:0.80.3'
}
```

And the current version of JDK that I'm using:
```terminal
java version "1.8.0_31"
Java(TM) SE Runtime Environment (build 1.8.0_31-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.31-b07, mixed mode)
```

I found an issue that seems like mine, but he is using Kotlin and I don't: https://github.com/realm/realm-java/issues/1179

As I said, it's an empty project and does nothing special. What could cause this problem? I'm a little lost. Thanks in advance!

**Comment by cmelchior** [comment]
Hi @jgonfer 
It is a known bug that happens if you have no Realm model classes (i.e. not any classes that extends RealmObject). We are already tracking this here: https://github.com/realm/realm-java/issues/860, but I agree this can be confusing when you first add Realm.

**Comment by cmelchior** [comment]
Just to note that this has been fixed in master when we merged https://github.com/realm/realm-java/pull/929 so should be available in our ```0.81-SNAPSHOT```

**Comment by jgonfer** [comment]
You're right. I've created a Realm model class to test it and the problem has been fixed. Thanks a lot @cmelchior :)
